src/tools/kdb/command.h

```diff
@@ -0,0 +1,260 @@
+
+/**
+ * @def GET_OPT_KEY
+ * @brief This macro retrieves the key with a given name from an options key set. This is the more general version that GET_OPTION_KEY in
+ * command files should use.
+ * @param options The options KeySet.
+ * @param key The name of the key to retrieve.
+ */
+#define GET_OPT_KEY(options, key) ksLookupByName (options, key, 0)
+
+/**
+ * @def GET_OPT
+ * @brief This macro retrieves the string value of the key with a given name from an options key set. This is the more general version that
+ * GET_OPTION in command files should use.
+ * @param options The options KeySet.
+ * @param key The name of the key to retrieve.
+ */
+#define GET_OPT(options, key) keyString (GET_OPT_KEY (options, key))
+
+/**
+ * @def HAS_ERR
+ * @brief This macro checks if an error/reason meta-key exists in the given key.
+ * @param errorKey The key to check for an error/reason meta-key.
+ */
+#define HAS_ERR(errorKey) keyGetMeta (errorKey, "error/reason") != NULL
+
+/**
+ * @def HAS_ERR_CODE
+ * @brief This macro checks if an error/number meta-key exists in the given key.
+ * @param errorKey The key to check for an error/number meta-key.
+ */
+#define HAS_ERR_CODE(errorKey) keyGetMeta (errorKey, "error/number") != NULL
+
+/**
+ * @def GET_ERR
+ * @brief This macro retrieves the string value of the error/reason meta-key in the given key.
+ * @param errorKey The key to get the error/reason from.
+ */
+#define GET_ERR(errorKey) keyString (keyGetMeta (errorKey, "error/reason"))
+
+/**
+ * @def GET_ERR_CODE
+ * @brief This macro retrieves the string value of the error/number meta-key in the given key.
+ * @param errorKey The key to get the error/number from.
+ */
+#define GET_ERR_CODE(errorKey) keyString (keyGetMeta (errorKey, "error/number"))
+
+/**
+ * @def GET_ERR_DESC
+ * @brief This macro retrieves the string value of the error/description meta-key in the given key.
+ * @param errorKey The key to get the error/description from.
+ */
+#define GET_ERR_DESC(errorKey) keyString (keyGetMeta (errorKey, "error/description"))
+
+/**
+ * @def COMMAND_BASE_KEY
+ * @brief This macro generates the base key name for a given command. Used in command files for defining GET_OPTION_KEY and GET_OPTION.
+ * @param name The full keyname for a command.
+ */
+#define COMMAND_BASE_KEY(name) CLI_BASE_KEY "/" name
+
+/**
+ * @def COMMAND_SPEC_KEY
+ * @brief This macro generates the specification key name for a given command. Used in command files when setting the spec.
+ * @param name The name of the command.
+ */
+#define COMMAND_SPEC_KEY(name) "spec:" COMMAND_BASE_KEY (name)
+
+/**
+ * @def GET_BASIC_OPTIONS
+ * @brief This macro retrieves the values of the basic options from an options key set into local variables. This has to be used in the
+ * context of a command, so GET_OPTION_KEY has to be defined.
+ */
+#define GET_BASIC_OPTIONS                                                                                                                  \
```

Comment:
Needs a little explaination what `GET_OPTION_KEY` actually should do.

```suggestion
/**
 * @def GET_BASIC_OPTIONS
 * @brief This macro retrieves the values of the basic options from an options key set into local variables.
 * This has to be used in the context of a command, so #GET_OPTION_KEY has to be defined.
 *
 * #GET_OPTION_KEY should be `#define GET_OPTION_KEY(options, name)` and evaluate to a `Key *` given a `KeySet * options` and a `char * name`.
 * 
 */
#define GET_BASIC_OPTIONS                                                                                                                  \
```